CONTRIBUTING.md

```diff
@@ -0,0 +1,111 @@
+# Contribute to AMReX
+
+Here the workflow we use for AMReX development.
+
+## Git workflow
+
+AMReX uses [git](https://git-scm.com) for version control. If you
+are new to git, you can follow one of these tutorials:
+- [Learn git with bitbucket](https://www.atlassian.com/git/tutorials/learn-git-with-bitbucket-cloud)
+- [git - the simple guide](http://rogerdudler.github.io/git-guide/)
+
+### Make your own fork and create a branch on it
+
+The basic workflow is:
+- fork the main repo (or update it if you already created it);
+- Implement your changes and push them on a new branch `<branch_name>` on
+your fork;
+- Create a Pull Request from branch `<branch_name>` on your fork to branch
+`development` on the main AMReX repository.
+
+First, let us setup your local git repo. Make your own fork of the main
+(`upstream`) repository:
+on the [AMReX Github page](https://github.com/atmyers/amrex), press the
+fork button. Then, you can execute:
+```
+# Clone your fork on your local computer. You can get this address on your fork's Github page.
+git clone --branch development https://github.com/<myGithubUsername>/amrex.git
+
+# Add a new remote for the main AMReX repo, to remain up-to-date.
+git remote add upstream https://github.com/AMReX-Codes/amrex
+
+# For safety, disable pushing directly to upstream (only relevant if you have write access)
+git remote set-url --push upstream no_push
+```
+Now you are free to play with your fork (for additional information, you can visit the
+[Github fork help page](https://help.github.com/en/articles/fork-a-repo)).
+
+> Note: you do not have to re-do the setup above every time.
+> Instead, in the future, you need to update the `development` branch
+> on your fork with
```

Comment:
Perhaps we can consider recommending that they make their development branch on their local system point to `upstream` instead of their fork, so that they don't have to remember to add that `upstream` source branch every time (I know I would constantly forget).

Reply:
Meaning, on the `development` branch, do `git branch -u upstream/development`

Reply:
Indeed. This would make your recommendation equivalent to Weiqun's (he advocated cloning upstream so you get the right development and master, and then pushing feature branches to one's fork), but yours is a little easier.

Reply:
Okay, I agree and have made this change.